pico/tools/src/main/java/io/helidon/pico/tools/ModuleInfoDescriptor.java

```diff
@@ -0,0 +1,633 @@
+
+    /**
+     * The default module name (i.e., "unnamed").
+     */
+    String DEFAULT_MODULE_NAME = "unnamed";
+
+    /**
+     * The default suffix on the module name for a test module. This suffix will be appended to the module name
+     * of the main module.
+     */
+    String DEFAULT_TEST_SUFFIX = "test";
+
+    /**
+     * The base module-info name.
+     */
+    String MODULE_INFO_NAME = "module-info";
+
+    /**
+     * The java module-info name.
+     */
+    String DEFAULT_MODULE_INFO_JAVA_NAME = MODULE_INFO_NAME + ".java";
+
+    /**
+     * The resource providing the module-info template.
+     */
+    String SERVICE_PROVIDER_MODULE_INFO_HBS = "module-info.hbs";
+
+
+    /**
+     * Used to declare the preferred ordering of the items in the module-info.
+     */
+    enum Ordering {
+
+        /**
+         * Little or no attempt is made to preserve comments, loaded/created ordering is arranged top-down.
+         */
+        NATURAL,
+
+        /**
+         * Attempt is preserve comments and natural, loaded/created ordering is arranged top-down.
+         */
+        NATURAL_PRESERVE_COMMENTS,
+
+        /**
+         * Little or no attempt is made to preserve comments, ordering is arranged sorted by the target class or package.
+         */
+        SORTED
+
+    }
+
+    /**
+     * The module name.
+     *
+     * @return the module name
+     */
+    @ConfiguredOption(DEFAULT_MODULE_NAME)
+    String name();
+
+    /**
+     * The template name to apply. The default is {@link io.helidon.pico.tools.TemplateHelper#DEFAULT_TEMPLATE_NAME}.
+     *
+     * @return the template name
+     */
+    @ConfiguredOption(TemplateHelper.DEFAULT_TEMPLATE_NAME)
+    String templateName();
+
+    /**
+     * The header (i.e., copyright) comment - will appear at the very start of the output.
+     *
+     * @return the header comment
+     */
+    Optional<String> headerComment();
+
+    /**
+     * The description comment - will appear directly above the module's {@link #name()}.
+     *
+     * @return the description comment
+     */
+    Optional<String> descriptionComment();
+
+    /**
+     * The ordering applied.
+     *
+     * @return the ordering
+     */
+    @ConfiguredOption("Ordering.NATURAL")
+    Ordering ordering();
+
+    /**
+     * The items contained by this module-info.
+     *
+     * @return the items
+     */
+    @Singular
+    List<Item> items();
+
+    /**
+     * Returns true if this module info is unnamed.
+     *
+     * @return true if this module is unnamed
+     */
+    default boolean isUnnamed() {
+        return DEFAULT_MODULE_NAME.equals(name());
+    }
+
+    /**
+     * Loads the {@code module-info} descriptor given its source file location.
+     *
+     * @param file the source file location
+     * @return the module-info descriptor
+     * @throws io.helidon.pico.tools.ToolsException if there is any exception encountered
+     */
+    static ModuleInfoDescriptor uncheckedLoad(File file) {
```

Comment:
1. we use `Path`, not `File` in Helidon
2. we try not to have checked exceptions in public APIs (except where required by implemented interface), so only use unchecked exceptions. There is an `UncheckedIoException` available in Java you can use here
3. static factory methods should be named `create`

Reply:
1. will change File -> Path.
2. I am using ToolsException (which is unchecked) for all forms of exceptions here including IO related.  If you prefer I can separate out a ToolsException (e.g., badly formatted file, or unhandled parsing scenario involving inner comments, etc) from a pure IO related exception.  LMK..
3. 3. will change to create.